src/transformStateWithClones.js

```diff
@@ -7,7 +7,25 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArray = [];
+  let copyState = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(copyState, action.extraData);
+      resultArray.push(Object.assign({}, copyState));
+    } else if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete copyState[key];
+      }
+      resultArray.push(Object.assign({}, copyState));
+    } else if (action.type === 'clear') {
+      copyState = {};
+      resultArray.push({});
+    }
```

Comment:
[CODE STYLE]: Include a `default` case in your `switch` statement to handle any unexpected `action.type` values. This is important for error handling and code robustness.